src/lib/probe.c

```diff
@@ -38,6 +38,94 @@ static char *cbm_devnode_to_devpath(dev_t dev)
         return realpath(c, NULL);
 }
 
+/**
+ * Convert a sysfs dev file to the target device path
+ */
+static char *cbm_dev_file_to_devpath(const char *devfile)
+{
+        int fd = 0;
+        unsigned int dev_major, dev_minor = 0;
+        char read_buf[PATH_MAX] = { 0 };
+        ssize_t size = -1;
+
+        /* Read the dev file */
+        fd = open(devfile, O_RDONLY | O_NOCTTY | O_CLOEXEC);
+
+        if (fd < 0) {
+                return NULL;
+        }
+
+        size = read(fd, read_buf, sizeof(read_buf));
+        close(fd);
+        if (size < 1) {
+                return NULL;
+        }
+
+        if (sscanf(read_buf, "%u:%u", &dev_major, &dev_minor) != 2) {
+                return NULL;
+        }
+
+        return cbm_devnode_to_devpath(makedev(dev_major, dev_minor));
+}
+
+static char *cbm_get_luks_uuid(const char *part)
+{
+        autofree(char) *npath = NULL;
+        autofree(char) *dpath = NULL;
+        glob_t glo = { 0 };
+        blkid_probe blk_probe = NULL;
+        char *ret = NULL;
+        const char *value = NULL;
+
+        /* i.e. /sys/block/dm-1/slaves/dm-0/slaves/sdb1/dev */
+        if (asprintf(&npath, "/sys/block/%s/slaves/*/slaves/*/dev", part) < 0) {
+                DECLARE_OOM();
+                return NULL;
+        }
+
+        glob(npath, GLOB_DOOFFS, NULL, &glo);
+
+        if (glo.gl_pathc < 1) {
+                globfree(&glo);
+                return NULL;
+        }
+
+        dpath = cbm_dev_file_to_devpath(glo.gl_pathv[0]);
```

Comment:
Would you mind explaining to me as I know too little about LUKS or LVM why any device node that glob finds is going to be fine to use here?